### DISCUSSION TITLE
Custom function code response for pymodbus tcp simulator

### DISCUSSION THREAD

**Discussion by abijithshetty** [pymodbus-dev/pymodbus#1954] [Help] [answered]
Hi ,
Is it possible to make pymodbus simulator return exception to [inject exception] to function code 7 or 8 ? like on receiving request simulator should raise an exception of "illegal function code"

Thanks

**Comment by janiversen** [comment, marked as answer]
Yes read the documentation, the fastest way is to add a response tracer.

**Reply by abijithshetty** [reply]
thanks @janiversen , got some idea from "examples/server_hook.py"

**Reply by janiversen** [reply]
Good for you.
